src/BridgedRoom.ts

```diff
@@ -266,18 +278,18 @@ export class BridgedRoom {
                 emojiKeyName = emojiKeyName.substring(1, emojiKeyName.length - 1);
             }
         }
-        let client: WebClient = this.botClient;
-        const puppet = await this.main.clientFactory.getClientForUserWithId(this.SlackTeamId!, message.sender);
-        if (puppet) {
-            client = puppet.client;
-            // We must do this before sending to avoid racing
-            // Use the unicode key for uniqueness
-            this.addRecentSlackMessage(`reactadd:${relatesTo.key}:${puppet.id}:${event.slackTs}`);
+        const client = await this.getClientForRequest(message.sender);
+        if (!client) {
+            log.warn("No client to handle reaction");
+            return;
         }
+        // We must do this before sending to avoid racing
+        // Use the unicode key for uniqueness
+        this.addRecentSlackMessage(`reactadd:${relatesTo.key}:${client.id}:${event.slackTs}`);
 
         // TODO: This only works once from matrix if we are sending the event as the
         // bot user.
-        const res = await client.reactions.add({
+        const res = await client.client.reactions.add({
```

Comment:
`client.client.` is confusing.
I would prefer the outer wrapper client to be renamed to something that indicates what else the object does contain.
Of unwrap the object right away:
```
const { clientId: id, client } = getClientForRequest()
```

Reply:
Ack, I couldn't unwrap because it was null but I can fix the naming.